### PR TITLE
feat(retrieve card data) CHK 1828 cache value

### PR DESCRIPTION
**Pull request by Fernando-Granato**
#### List of Changes
👉🏻 Added logic to cache card data received from **NPG**.

#### Motivation and Context
The change was made to allow the caching of the values ​​returned by the NPG during card information retrieval.

#### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: src/main/java/it/pagopa/ecommerce/payment/methods/infrastructure/NpgSessionDocument.java [hunk 1]
```diff
@@ -4,11 +4,16 @@
...
 
 @RedisHash(value = "keys", timeToLive = 10 * 60)
 public record NpgSessionDocument(
         @NonNull @Id String sessionId,
-        @NonNull String securityToken
+        @NonNull String securityToken,
+        @Nullable String bin,
+        @Nullable String lastFourDigits,
+        @Nullable String expiringDate,
+        @Nullable String circuit
```
Comment (giovanniberti) [inline src/main/java/it/pagopa/ecommerce/payment/methods/infrastructure/NpgSessionDocument.java:16]: If possible I would make a separate structure and declare that as nullable, i.e.:
```java
public record CardData(
    String bin,
    String lastFourDigits,
    String expiringDate,
    String circuit
) {}
```

```suggestion
        @Nullable CardData cardData
```

File: src/test/java/it/pagopa/ecommerce/payment/methods/service/PaymentMethodServiceTests.java [hunk 2]
```diff
@@ -341,7 +344,21 @@ void shouldRetrieveCardDataForInvalidPaymentMethodId() {
...
+        String paymentMethodId = "paymentMethodId";
+        String sessionId = "sessionId";
+        PaymentMethod paymentMethod = TestUtil.getPaymentMethod();
+        PaymentMethodDocument paymentMethodDocument = TestUtil.getTestPaymentDoc(paymentMethod);
+        Mockito.when(paymentMethodRepository.findById(paymentMethodId)).thenReturn(Mono.just(paymentMethodDocument));
+        Mockito.when(npgSessionsTemplateWrapper.findById(sessionId)).thenReturn(Optional.empty());
+        StepVerifier.create(paymentMethodService.getCardDataInformation(paymentMethodId, any()))
+                .expectErrorMatches(e -> e instanceof SessionIdNotFoundException)
+                .verify();
+
```
Comment (pietro-tota) [inline src/test/java/it/pagopa/ecommerce/payment/methods/service/PaymentMethodServiceTests.java:357]: Add mockito verify assertions against redis wrapper and npgClient mock method calls
```suggestion
        Mockito.verify(npgSessionsTemplateWrapper, Mockito.times(1)).findById(any());
        Mockito.verify(npgSessionsTemplateWrapper, Mockito.times(0)).save(any());
        Mockito.verify(npgClient, Mockito.times(0)).getCardData(any(), any());
```

File: src/test/java/it/pagopa/ecommerce/payment/methods/service/PaymentMethodServiceTests.java [hunk 3]
```diff
@@ -341,7 +344,21 @@ void shouldRetrieveCardDataForInvalidPaymentMethodId() {
     }
 
     @Test
-    void shouldRetrieveCardDataWithSuccess() {
+    void shouldRetrieveCardDataForInvalidSessionId() {
```
Comment (pietro-tota) [inline src/test/java/it/pagopa/ecommerce/payment/methods/service/PaymentMethodServiceTests.java:347]:
```suggestion
    void shouldReturnErrorForInvalidSessionId() {
```

File: src/main/java/it/pagopa/ecommerce/payment/methods/infrastructure/NpgSessionDocument.java (hunk 1 above)
Comment (Fernando-Granato) [reply src/main/java/it/pagopa/ecommerce/payment/methods/infrastructure/NpgSessionDocument.java]: Fixed with 3bfc15f394be9c0352023039ae79845dfe119d29!

File: src/test/java/it/pagopa/ecommerce/payment/methods/service/PaymentMethodServiceTests.java (hunk 3 above)
Comment (Fernando-Granato) [reply src/test/java/it/pagopa/ecommerce/payment/methods/service/PaymentMethodServiceTests.java]: Fixed with b37aebcbbd927a6d9d76f8d64211e242b2fedf18

File: src/test/java/it/pagopa/ecommerce/payment/methods/service/PaymentMethodServiceTests.java (hunk 2 above)
Comment (Fernando-Granato) [reply src/test/java/it/pagopa/ecommerce/payment/methods/service/PaymentMethodServiceTests.java:357]: Fixed with ccf8910831947ede3edd4a71e5002d8aa12e7f1b

**Review by pietro-tota — APPROVED** [review]
🚀

**Review by giovanniberti — APPROVED** [review]

**Pull request merged**